### PR TITLE
[allocator.adaptor.members] Fix select_on_container_copy_construction

### LINKED ISSUE #518: [allocator.adaptor.members] Imprecise wording for scoped_allocator_adaptor::select_on_container_copy_construction
[allocator.adaptor.members]/17 says:

> from the result of calling `allocator_traits<A>::select_on_container_copy_construction()` on
> the corresponding allocator in `*this`.

That function takes an argument, so displaying it with an empty set of parentheses is misleading.

Saying calling it "on the corresponding allocator" doesn't clearly say that the corresponding allocator is passed as the argument to the function.

I suggest something like:

> from the result of calling `allocator_traits<A>::select_on_container_copy_construction` with
> the corresponding allocator in `*this` as the function argument.

### PR DESCRIPTION
**Pull request by jensmaurer**
The description was confusing objects and types.

Fixes #518

### CODE REVIEW

**Comment by jensmaurer** [comment]
@jwakely , after more than half a decade, it's about time this gets fixed. Please have a look.

File: source/utilities.tex [hunk 1]
```diff
@@ -13950,10 +13950,11 @@
...
 \returns
-A new \tcode{scoped_allocator_adaptor} object where each allocator \tcode{A} in the
-adaptor is initialized from the result of calling
-\tcode{allocator_traits<A>::select_on_container_copy_construction()} on the
-corresponding allocator in \tcode{*this}.
+A new \tcode{scoped_allocator_adaptor} object
+where each allocator \tcode{a} of type \tcode{A} within the adaptor
+is initialized with
+\tcode{allocator_traits<A>::select_on_container_copy_construction(a2)},
+where \tcode{a2} is the corresponding allocator in \tcode{*this}.
```
Comment (jwakely) [inline source/utilities.tex:13957]: I'm finding it hard not to read "each allocator `a` of type `A`" as meaning this only applies to the ones of type `A`, for some specific type `A`.

How about:


```suggestion
A new \tcode{scoped_allocator_adaptor} object
where each allocator \tcode{a} within the adaptor
is initialized with
\tcode{allocator_traits<A>::select_on_container_copy_construction(a2)},
where \tcode{A} is the type of \tcode{a} and \tcode{a2} is the corresponding allocator in \tcode{*this}.
```

Comment (jensmaurer) [reply source/utilities.tex]: You're right. Fixed.

**Review by jwakely — APPROVED** [review]

File: source/utilities.tex (hunk 1 above)
Comment (tkoeppe) [reply source/utilities.tex]: How about instead of `a` and `a2` we have `a1` and `a2` symmetrically?

Otherwise good, thanks!

Comment (jwakely) [reply source/utilities.tex]: That seems fine too.

I considered something fancy like `A`<sub>n</sub>, `a`<sub>n</sub> and `this->a`<sub>n</sub>, but it wouldn't be right as `this->a`<sub>n</sub> isn't actually a member of `*this`.

Comment (tkoeppe) [reply source/utilities.tex]: @jensmaurer: I'd somewhat prefer `a1`, `a2` over `a`, `a2`, but if you feel very strongly, we can do it your way. Please feel free to merge either version!

Comment (tkoeppe) [reply source/utilities.tex]: @jensmaurer: For your consideration, but fine to go in in principle.

Comment (jensmaurer) [reply source/utilities.tex]: @tkoeppe, change to a1, a2. Please double-check.

Comment (tkoeppe) [reply source/utilities.tex]: Thanks, I like it! Can we also make the type of `a1` be `A1`?

Comment (jensmaurer) [reply source/utilities.tex]: @tkoeppe, done.

Comment (tkoeppe) [reply source/utilities.tex]: Thank you!

**Pull request merged**